dcos/http.py

```diff
@@ -182,16 +205,7 @@ def request(method,
     if 'headers' not in kwargs:
         kwargs['headers'] = {'Accept': 'application/json'}
 
-    if verify is None and constants.DCOS_SSL_VERIFY_ENV in os.environ:
-        verify = os.environ[constants.DCOS_SSL_VERIFY_ENV]
-        if verify.lower() == "true":
-            verify = True
-        elif verify.lower() == "false":
-            verify = False
-
-    # Silence 'Unverified HTTPS request' and 'SecurityWarning' for bad certs
-    if verify is not None:
-        silence_requests_warnings()
+    verify = _verify_ssl(verify)
```

Comment:
I don't want to move `_verify_ssl` to `_request` because in the case of auth, we might call `_request` multiple times and we'll be doing unnecessary calls. But ill remove the side effects by moving the call to  `silence_requests_warning` outside of `_verify_ssl`